CMake/Version.cmake

```diff
@@ -0,0 +1,86 @@
+                       CMAKE_SYSTEM_NAME STREQUAL "OpenBSD")
+                        set(_ver_type "sunos")
+                elseif(CMAKE_SYSTEM_NAME STREQUAL "SunOS")
+                        # This is very simplified but should be enough
+                        string(SUBSTRING 0 1 _v "${CMAKE_SYSTEM_VERSION}")
+                        if(_v STREQUAL "5" OR _v STREQUAL "6")
+                                set(_ver_type "linux")
+                        else()
+                                set(_ver_type "sunos")
+                        endif()
+                elseif(CMAKE_SYSTEM_NAME STREQUAL "SCO_SV" OR
+                       CMAKE_SYSTEM_NAME STREQUAL "UnixWare" OR
+                       CMAKE_SYSTEM_NAME STREQUAL "UNIX_SV" OR
+                       CMAKE_SYSTEM_NAME STREQUAL "XENIX" OR
+                       QNXNTO)
+                        set(_ver_type "sco/qnx")
+                elseif(APPLE)
+                        set(_ver_type "darwin")
+                endif()
+        endif()
+
+        set(${_out_version_type_name} "${_ver_type}" PARENT_SCOPE)
+endfunction()
+
+# Take the libtool-specific version info in form CURRENT:REVISION:AGE
+# and try to translate it to CMakes' VERSION / SOVERSION properties
+function(parse_versioninfo _versioninfo_string _out_version_name _out_soversion_name)
+        string(REPLACE ":" ";" _versioninfo_components ${_versioninfo_string})
+        list(LENGTH _versioninfo_components _len)
+        if(${_len} LESS 3)
+                message(FATAL_ERROR "Version string '${_versioninfo_string}' is not a valid libtool value.")
+        endif()
+
+        list(GET _versioninfo_components 0 _current)
+        list(GET _versioninfo_components 1 _revision)
+        list(GET _versioninfo_components 2 _age)
+
+        determine_version_type(_version_type)
+
+        if(_version_type STREQUAL "linux")
+                math(EXPR _major ${_current}-${_age})
+                set(${_out_version_name} ${_major}.${_age}.${_revision} PARENT_SCOPE)
+                set(${_out_soversion_name} ${_major} PARENT_SCOPE)
+        elseif(_version_type STREQUAL "sco/qnx")
+                set(${_out_version_name} ${_current} PARENT_SCOPE)
+                set(${_out_soversion_name} ${_current} PARENT_SCOPE)
+        elseif(_version_type STREQUAL "sunos")
+                set(${_out_version_name} ${_current} PARENT_SCOPE)
+                set(${_out_soversion_name} ${_current}.${_revision} PARENT_SCOPE)
+        elseif(_version_type STREQUAL "darwin")
+                math(EXPR _minor_current ${_current}+1)
+                set(${_out_version_name} ${_minor_current}.${_revision} PARENT_SCOPE)
+                set(${_out_soversion_name} ${_minor_current} PARENT_SCOPE)
+        elseif(_version_type STREQUAL "windows" OR _version_type STREQUAL "cygwin")
+                math(EXPR _major ${_current}-${_age})
+                set(${_out_version_name} ${_major} PARENT_SCOPE)
```

Comment:
cygwin should get the version info in the soname, all the cygwin dlls have name-version

Reply:
...and the build currently produces "libcurl-4.dll". The logic to produce the right file name is in lib/CMakeLists.txt:87
That being said; It seems that if I set the soversion, the target gets the "-version" suffix automatically, so I'll change it.